packages/extension-youtube/src/youtube.ts

```diff
@@ -0,0 +1,118 @@
+import { Node, mergeAttributes } from '@tiptap/core'
+
+import { getEmbedURLFromYouTubeURL, isValidYouTubeUrl } from './utils'
+
+export interface YoutubeOptions {
+  inline: boolean;
+  width: number;
+  height: number;
+  controls: boolean;
+  nocookie: boolean;
+  allowFullscreen: boolean;
+  HTMLAttributes: Record<string, any>,
+}
+
+declare module '@tiptap/core' {
+  interface Commands<ReturnType> {
+    youtube: {
+      /**
+       * Insert a youtube video
+       */
+      setYoutubeVideo: (options: { src: string, width?: number, height?: number, start?: number }) => ReturnType,
+    }
+  }
+}
+
+export const YouTube = Node.create<YoutubeOptions>({
```

Comment:
I think we should unify that. YouTube's brand is written in CapitalCase, but I would suggest to write only the first letter of every word in capital letters. Because YoutubeDownloader do it as well. 😁